package/network/services/cifsd-tools/files/cifsd.init

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh /etc/rc.common
+
+START=98
+
+# FIXME: We get a router crash/restart using procd via 'cifsd -n', because of the two worker, manager processes ?
+
+start(){
+
+	if [ ! -f /etc/cifs/smb.conf ]; then
+		logger -t 'cifsd' "missing config /etc/cifs/smb.conf, needs to-be created manually!"
+		exit 1
+	fi
+
+	[ -f /tmp/cifsd.lock ] && rm /tmp/cifsd.lock
+	
+	# try remove again before start
+	if (lsmod | grep cifsd &>/dev/null); then
+		rmmod cifsd &>/dev/null
+	fi
+	modprobe cifsd 2>/dev/null
+	if !(lsmod | grep cifsd &>/dev/null); then
+		logger -t 'cifsd' "modprobe of cifsd module failed, cant start cifsd!"
+		exit 1
+	fi
+
+	logger -t 'cifsd' "Starting CIFS/SMB userspace service."
+	/usr/bin/env LANG=en_US.UTF-8 /usr/sbin/cifsd
```

Comment:
It would be nice if you could use procd to manage this daemon, procd also takes crase to restart it if it crashes and that it gets shut down later on.

Reply:
See my comment at the ini file, atm its incompatible with procd and i would get restarts. It seems the way the 2 helper threads operate causes some problems with procd.

Reply:
> See my comment at the ini file, atm its incompatible with procd and i would get restarts. It seems the way the 2 helper threads operate causes some problems with procd.

@Andy2244 Would be nice to get this procd issue fixed, instead of this workarounds. Could you please report this procd issue to `openwrt-devel@lists.openwrt.org` or at http://bugs.openwrt.org ?